server/mediaserver_test.go

```diff
@@ -435,9 +439,38 @@ func TestRegisterConnection(t *testing.T) {
 	mid := core.SplitStreamIDString(t.Name()).ManifestID
 	strm := stream.NewBasicRTMPVideoStream(string(mid))
 
-	// Should return an error if missing node storage
-	drivers.NodeStorage = nil
+	// Switch to on-chain mode
+	c := &eth.MockClient{}
+	addr := ethcommon.Address{}
+	s.LivepeerNode.Eth = c
+
+	// Should return an error if in on-chain mode and fail to get sender deposit
+	c.On("Account").Return(accounts.Account{Address: addr})
+	c.On("Senders", addr).Return(nil, nil, nil, errors.New("Senders error")).Once()
```

Comment:
Just a note that I don't find this particular mocking API too nice. The style is a bit too indirect for me. Much prefer post-test assertions (rather than inverting the flow of logic with expects) and all those `Return` parameters look unwieldy.

I know we're still experimenting with different testing styles though -- and I really haven't looked at what else might be available so I'm just complaining without providing constructive alternatives ATM.

This isn't a change request, just noting some personal impressions around this mocking API.

Reply:
Yeah I hear you. Definitely up for finding a solid mocking style that everyone is happy with - let's add it as a discussion topic to refactoring week!